library/x509_crt.c

```diff
@@ -1435,32 +1571,179 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 }
 #endif /* MBEDTLS_FS_IO */
 
+/*
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ *
+ * HardwareModuleName ::= SEQUENCE {
+ *                           hwType OBJECT IDENTIFIER,
+ *                           hwSerialNum OCTET STRING }
+ *
+ * NOTE: we currently only parse and use otherName of type HwModuleName,
+ * as defined in RFC 4108.
+ */
+int mbedtls_x509_get_other_name( const mbedtls_x509_sequence *subject_alt_name,
+                                 mbedtls_x509_san_othername *other_name )
+{
+    int ret;
+    size_t len;
+    unsigned char *p = subject_alt_name->buf.p;
+    const unsigned char *end = p + subject_alt_name->buf.len;
+    mbedtls_x509_buf cur_oid = { 0, 0, NULL };
```

Comment:
Please see my other comment. I think that preferably we should avoid initialising structs like this because we have to update all the individual initialization statements whenever a new field is added to mbedtls_x509_buf

Reply:
ok

Reply:
Thanks for the reply. I am aware that we have such initializations, but that does not mean we should add more. Also, you can initialize the struct in a field-by-field way, though I suppose that can be equally problematic.